extensions/replication/utils/SetupReplication.js

```diff
@@ -103,46 +108,59 @@ class SetupReplication {
      * @param {RoundRobin} params.target.hosts - destination hosts
      * @param {String} [params.target.transport] - transport protocol for
      *   target (http/https)
+     * @param {Boolean} params.isTargetExternal - whether target bucket is on
+     *   an external location
+     * @param {String} [params.siteName] - the site name where the target bucket
+     *   exists, if the target is on an external location
      * @param {Object} params.log - werelogs request logger object
      */
     constructor(params) {
-        const { source, target, log } = params;
+        const { source, target, isTargetExternal, siteName, log } = params;
         this._log = log;
         this._sourceBucket = source.bucket;
         this._targetBucket = target.bucket;
+        this._isTargetExternal = isTargetExternal;
+        this._siteName = siteName;
         this.destHosts = target.hosts;
         const destHost = this.destHosts.pickHost();
         this._s3Clients = {
             source: _setupS3Client(source.transport,
                 `${source.s3.host}:${source.s3.port}`, source.credentials),
-            target: _setupS3Client(target.transport,
+            target: isTargetExternal ? undefined :
+                _setupS3Client(target.transport,
                 `${destHost.host}:${destHost.port}`, target.credentials),
         };
         this._iamClients = {
             source: _setupIAMClient('source', source.transport,
                 `${source.vault.host}:${source.vault.adminPort}`,
                 source.credentials),
             // XXX use target port through nginx gateway
-            target: _setupIAMClient('target', target.transport,
-                `${destHost.host}:${source.vault.adminPort}`,
+            target: isTargetExternal ? undefined : _setupIAMClient('target',
+                target.transport, `${destHost.host}:${source.vault.adminPort}`,
                 target.credentials),
         };
     }
 
     checkSanity(cb) {
         return async.waterfall([
             next => this._isValidBucket('source', next),
-            next => this._isValidBucket('target', next),
+            next => (this._isTargetExternal ? next() :
+                this._isValidBucket('target', next)),
```

Comment:
It would be great if sanity check functions `isValidBucket`, `isVersioningEnabled` etc. could also check external backends! I understand why it's skipped here because the implementation is different (has to go through the source with probably differing bits), but is there a plan to do this in the future? Similarly, we could also pre-create buckets and enable versioning :)

Reply:
We wanted to leave AWS buckets up to the user to handle, and not modify them in the script. I've documented that fact accordingly here: https://github.com/scality/backbeat/pull/128/files#diff-c86568b6c2391c54b2749733f7d7ceebR90.